doc/specs/vulkan/chapters/fundamentals.txt

```diff
@@ -754,14 +766,14 @@ commands, which will silently ignore these values.
 [[fundamentals-validusage-pointers]]
 ==== Valid Usage for Pointers
 
-Any parameter that is a pointer must: either be a valid pointer, or if
-_explicitly called out in the specification_, can: be `NULL`.
+A Valid Usage section always explicitly states if a command parameter or a
+structure member is required to be a valid pointer.
```

Comment:
I actually missed the subtlety when I reviewed this the first time, but this kind of implies that if we "miss" putting a "must be a valid pointer" in the VU, then it doesn't have to be valid even if the specification text says otherwise, which doesn't make any sense. I don't think we should take this paragraph change.

Reply:
That is how it actually works though.
If the specification text says otherwise, such specification text is or should be moved to a VU section anyway.

Reply:
I don't disagree that it *should* be moved to the VU, and that any time we see something that isn't in a VU it should definitely move TO a VU, but we have not committed to saying that "If a VU isn't there it's fine" because we don't have a good process for ensuring that is the case.

IMO this is a massive over-commitment from what is true right now, though I would certainly love to be able to put that statement in a future spec. It just suggests something that is not currently guaranteed to be accurate, as much as we want it to be.

Reply:
NB: I have a proposal sitting around that makes it more tenable to add something like this but it's a ways off completion at the moment I'm afraid.

Reply:
>IMO this is a massive over-commitment from what is true right now

I don't really see it.

Firstly in this case it seems the script is fully handling it.
And if the author removes it from the script, he is fully responsible to add relevant VU.

By this statement I am not really commiting to something new, more like fixing the situation that sometimes pointers are allowed to be invalid (bit of an oxymoron really). I could say, they can be invalid, if explicitly stated in the spec, but that would be a lie.

So my only option here is either to add the statement as above, or pretty much remove the chapter. And removing the chapter would have exactly the same semantic meaning as the added statement, so not much a choice.

Reply:
> So my only option here is either to add the statement as above, or pretty much remove the chapter. And removing the chapter would have exactly the same semantic meaning as the added statement, so not much of a choice.

That's kind of extreme, but I see your point.

I think the problem I have is literally the strength of the statement, it's like asserting there will never be bugs. Also you've sort of changed the tone of the paragraph so it doesn't really fit with the rest of them. I think a reword would make me less uneasy, how about:

> Any parameter that is a pointer must: be a valid pointer only if it is explicitly called out by a Valid Usage statement.

Reply:
> [...], it's like asserting there will never be bugs.

Heh, that's bit of a circular. If there was a bug, that would make the statement also a bug. When the bug is fixed, it would make the statement true again :p 

Well, I think it is a healthy assertion throughout the specification that it is bug free, even if not literally true due to a human error. I would be more worried if the spec said: "There may: be unknown amount of bugs in this specification. Enjoy!" 😆 

>I think a reword would make me less uneasy, how about:
>
>>Any parameter that is a pointer must: be a valid pointer only if it is explicitly called out by a Valid Usage statement.

I am fine with that, as I see absolutely no difference in it for my purposes.